### PR TITLE
Refactor config out of the main package

### PR DESCRIPTION
**Pull request by jbydeley**
Fix for #18 

- Rename beaconConfig to Config
- Create config package
  - Add tests and test data
- Create errors package
  - Allow easy error constants

### CODE REVIEW

File: errors/errors.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package errors
+
+import "fmt"
+
+// Error allows us to create constant errors
```
Comment (teesloane) [inline errors/errors.go]: I'm not sure if this error file implements the  [same](https://dave.cheney.net/2016/04/07/constant-errors) concepts as you referred to earlier; but I think if it is, we should put the link to the blog post in the code in a comment for those who are new to Go / interested. 

But, I also might just be projecting my own lack of understanding of errors so far into the code. What do you think?

File: main.go [hunk 2]
```diff
@@ -23,6 +23,8 @@ import (
 	"io/ioutil"
 	"log"
 	"os"
+
+	"github.com/jbydeley/beacon/config"
```
Comment (teesloane) [inline main.go]: I'm not super familiar with imports in Golang but when this merges to `the-heap`'s github account will the path for this need to be updated?

File: config/config_test.go [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+	for _, tc := range tt {
+		path := path.Join("testdata", tc.Path)
+		_, err := LoadFile(path)
+		if err != tc.Err {
+			t.Fatalf("LoadFile(%v) returned an error (%v)", path, err)
+		}
+	}
+}
+
+func Test_LoadFile_MissingFile(t *testing.T) {
```
Comment (teesloane) [inline config/config_test.go:37]: Excellent work / good thinking for test cases!

**Review by teesloane — APPROVED** [review]
This looks great to me — if you feel like the comments I left are kind of extraneous (ie the questions about importing files) feel free to respond but still merge anyway :)

File: errors/errors.go (hunk 1 above)
Comment (jbydeley) [reply errors/errors.go]: Sure, I like that idea. I'll add it shortly.

File: main.go (hunk 2 above)
Comment (jbydeley) [reply main.go]: Yes, I should update it before merging and probably just branch off this repo directly next time. Good catch!

File: errors/errors.go (hunk 1 above)
Comment (teesloane) [reply errors/errors.go]: Thanks!

**Comment by teesloane** [comment]
Looks great. Let's ship ship ship!

**Pull request merged**